api/src/main/java/jakarta/data/Repository.java

```diff
@@ -0,0 +1,329 @@
+ * <td>Jakarta Persistence providers limit the maximum to <code>Integer.MAX_VALUE</code></td></tr>
+ *
+ * <tr valign="top"><td><code>existsBy...</code></td>
+ * <td><code>boolean</code>, <code>Boolean</code></td>
+ * <td></td></tr>
+ *
+ * <tr valign="top"><td><code>findBy...</code></td>
+ * <td><code>E</code>,
+ * <br><code>Optional&lt;E&gt;</code></td>
+ * <td>For queries returning a single item (or none)</td></tr>
+ *
+ * <tr valign="top"><td><code>findBy...</code></td>
+ * <td><code>E[]</code>,
+ * <br><code>Iterable&lt;E&gt;</code>,
+ * <br><code>Stream&lt;E&gt;</code>,
+ * <br><code>Collection&lt;E&gt;</code></td>
+ * <td></td></tr>
+ *
+ * <tr valign="top"><td><code>findBy...</code></td>
+ * <td><code>Collection</code> subtypes</td>
+ * <td>The subtype must have a public default constructor and support <code>addAll</code> or <code>add</code></td></tr>
+ *
+ * <tr valign="top"><td><code>findBy...</code></td>
+ * <td><code>Page&lt;E&gt;</code>, <code>Iterator&lt;E&gt;</code></td>
+ * <td>For use with pagination</td></tr>
+ *
+ * <tr valign="top"><td><code>findBy...</code></td>
+ * <td><code>LinkedHashMap&lt;K, E&gt;</code></td>
+ * <td>Ordered map of Id attribute value to entity</td></tr>
+ *
+ * <tr valign="top"><td><code>save(E)</code></td>
+ * <td><code>E</code>,
+ * <br><code>void</code>, <code>Void</code></td>
+ * <td>For saving a single entity.</td></tr>
+ *
+ * <tr valign="top"><td><code>save(E...)</code>,
+ * <br><code>save(Iterable&lt;E&gt;)</code>,
+ * <br><code>save(Stream&lt;E&gt;)</code></td>
+ * <td><code>void</code>, <code>Void</code>,
+ * <br><code>E[]</code>,
+ * <br><code>Iterable&lt;E&gt;</code>,
+ * <br><code>Stream&lt;E&gt;</code>,
+ * <br><code>Collection&lt;E&gt;</code>
+ * <br><code>Collection</code> subtypes</td>
+ * <td>For saving multiple entities.
+ * <br>Collection subtypes must have a public default constructor
+ * and support <code>addAll</code> or <code>add</code></td></tr>
+ * </table>
+ * <p>
+ *
+ * <h3>Parameters to Repository Methods</h3>
+ *
+ * The parameters to a repository method correspond to the conditions that are
+ * defined within the name of repository method (see reserved keywords above),
+ * in the same order specified.
+ * Most conditions, such as <code>Like</code> or <code>LessThan</code>,
+ * correspond to a single method parameter. The exception to this rule is
+ * <code>Between</code>, which corresponds to two method parameters.<p>
+ *
+ * After all conditions are matched up with the corresponding parameters,
+ * the remaining repository method parameters are used to enable other
+ * capabilities such as pagination and sorting.<p>
+ *
+ * TODO write this section after pagination and sorting and other aspects
+ * are discussed and defined.<p>
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Repository {
+    /**
+     * Entity class. By default, detect automatically.
+     */
+    Class<?> value() default void.class;
```

Comment:
:question: do we need this value's field?

Reply:
> ❓ do we need this value's field?

There needs to be some way of identifying the type of entity for a repository for cases where the only references to the entity type are parameterized.  The following, for example, won't work for solutions that process the annotations at run time because it has no way of knowing that it needs to look for the location in the database that stores Car entities,

```
@Repository
public interface CarRepository extends DataRepository<Car, Long>{
    List<Car> findByMakeAndModel(String brand, String vehicleModel);
}
```

One way to allow for that is to allow the entity type to be optionally specified here,

```
@Repository(Car.class)
public interface CarRepository extends ...
```

But there are plenty of other options that could be chosen instead, such as having a separate annotation for it, or requiring the user to compose at least one method that accepts or returns the entity type unparameterized, and so forth.  Do you have a preference?

Reply:
We can do it with the annotation process with this way:

```java
    static List<String> findParameters(TypeMirror repository) {
        if (repository instanceof DeclaredType) {
            DeclaredType declaredType = (DeclaredType) repository;
            return declaredType.getTypeArguments().stream()
                    .map(TypeMirror::toString)
                    .collect(Collectors.toList());
        }
        return Collections.emptyList();
    }
```
Where the first parameter is the first element on the generic.
Anyway, we'll go further on it in the next discussion/issue about the ```CrudRepository```

Reply:
I don't see a way to use TypeMirror at run time, however, I looked into this further and it appears that ParameterizedType.getActualTypeArguments would be able to access the type parameters at run time, so I've updated the pull to remove the entity class from the annotation and its documentation.